packages/ember-metal/lib/namespace_search.js

```diff
@@ -184,6 +184,9 @@ function calculateToString(target) {
     }
     let superclass = target;
     do {
+      if (superclass === null) {
```

Comment:
If we add `Object.prototype` check (mentioned in another comment) we can remove the explicit `null` check..